src/backend/controllers/book.js

```diff
@@ -14,38 +14,65 @@ export const searchBooks = async (req, res) => {
 
   // I am not sure the below SQL is correct or not
   try {
-    const booksQuery = await db.query(`
-      SELECT b.ISBN, b.Title, b.PublisherName, b.AuthorName, MIN(ub.AskingPrice) AS LowestPrice, MAX(ub.AskingPrice) AS HighestPrice 
-      FROM BOOK AS b
-        JOIN USEDBOOK AS ub ON ub.bookID = b.ISBN
-        LEFT JOIN TEXTBOOK AS tb ON tb.bookID = b.ISBN
-        LEFT JOIN COURSEDEPT AS cd ON cd.serialNumber = tb.serialNumber AND cd.semester = tb.semester
-        LEFT JOIN COURSE AS c ON c.serialNumber = tb.serialNumber AND c.semester = tb.semester
-      WHERE LOWER(b.Title) LIKE LOWER('%${bookName || ""}%')
-      AND LOWER(b.AuthorName) LIKE LOWER('%${authorName || ""}%')
-      ${deptCode ? `AND cd.departmentCode = '${deptCode}'` : ``}
-      ${
-        courseName
-          ? `AND LOWER(c.courseName) LIKE LOWER('%${courseName}%')`
-          : ``
-      }
-      AND b.ISBN LIKE '%${isbn || ""}%'
-      GROUP BY b.ISBN
-    `);
+    const bookQueryValues = [
+      bookName ? "%" + bookName + "%" : "%",
+      authorName ? "%" + authorName + "%" : "%",
+    ];
+    let index = 3;
+    const indices = [];
+    if (deptCode) {
+      bookQueryValues.push(deptCode);
+      indices["deptCode"] = index++;
+    }
+    if (courseName) {
+      bookQueryValues.push("%" + courseName + "%");
+      indices["courseName"] = index++;
+    }
+    bookQueryValues.push(isbn ? "%" + isbn + "%" : "%");
+    indices["isbn"] = index++;
+
```

Comment:
Why does index start with 3 rather than 2

Reply:
Parameter indices in parameterized queries start from 1. See [https://node-postgres.com/features/queries](https://node-postgres.com/features/queries).

Reply:
ok